aiidalab_widgets_base/process.py

```diff
@@ -701,10 +708,10 @@ def __init__(self, callbacks=None, on_sealed=None, timeout=None, **kwargs):
 
         super().__init__(**kwargs)
 
-    @traitlets.observe("process")
+    @traitlets.observe("process_uuid")
     def _observe_process(self, change):
-        process = change["new"]
-        if process is None or process.id != getattr(change["old"], "id", None):
+        process_uuid = change["new"]
+        if process_uuid is None or process_uuid != change["old"]:
```

Comment:
Can we clean this up a bit to reduce the indentation levels? The first if statement is super confusing to me.

If I understood the logic correctly, this should work?
```suggestion
        if process_uuid == change["old"]:
            return
        # stop thread (if running)
        if self._monitor_thread is not None:
            with self._monitor_thread_lock:
                self._monitor_thread_stop.set()
                self._monitor_thread.join()
        
        if process_uuid is None:
            return

        with self.hold_trait_notifications():
            with self._monitor_thread_lock:
                self._monitor_thread_stop.clear()
                self._monitor_thread = Thread(
                    target=self._monitor_process, args=(process_uuid,)
                )
                self._monitor_thread.start()
        
```

Reply:
Yes, fully agree. I try to simplify it more. Please have a test of it. I'll do the test further more on qeApp.